File: packages/ios/Sources/Components/SearchBar/SearchBar.figma.swift
import SwiftUI
import Figma

struct KozmosSearchBarConnect: FigmaConnect {
    // The generic parameter is the trailing slot; a plain field has none.
    let component = KozmosSearchBar<EmptyView>.self
    let figmaNodeUrl = "https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=682-15111"

    @FigmaString("Placeholder Text")
    var placeholder: String = "Search..."

    @FigmaString("Value Text")
    var value: String = ""

    var body: some View {
        KozmosSearchBar(
            text: .constant(self.value),
            placeholder: self.placeholder
        )
    }
}

File: packages/android/src/main/java/com/kozmos/components/SearchBar/SearchBar.figma.kt
package com.kozmos.components.searchbar

import androidx.compose.runtime.Composable
import com.figma.code.connect.FigmaConnect
import com.figma.code.connect.FigmaProperty
import com.figma.code.connect.FigmaType

@FigmaConnect("https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=682-15111")
class KozmosSearchBarConnect {
    @FigmaProperty(FigmaType.Text, "Placeholder Text")
    val placeholder: String = "Search..."

    @FigmaProperty(FigmaType.Text, "Value Text")
    val value: String = ""

    @Composable
    fun ComponentExample() {
        KozmosSearchBar(
            value = value,
            onValueChange = {},
            placeholder = placeholder
        )
    }
}
